hazelcast/src/main/java/com/hazelcast/client/impl/protocol/AbstractCallableMessageTask.java

```diff
@@ -22,8 +22,8 @@
 /**
  * Base callable Message task.
  */
-public abstract class AbstractCallableMessageTask<CM extends ClientMessage>
-        extends AbstractMessageTask<CM> {
+public abstract class AbstractCallableMessageTask<P>
+        extends AbstractMessageTask<P> {
```

Comment:
Parameters are used by task implementers and they have to implement some abstract methods that are specific to that parameter impl so the actual guarding is in the task implementation. If we need to group these Parameters for some other reason we may define an interface in future. 

Side note: These Parameters actually STRUCT's but we don't have them in java yet. They are not actual objects but dummy parameter wrappers .